Adds a selection button to the toolbar of the 1.3.2 filer.

Updated for 1.3.3 CVS 2002-07-30 by Marcin Juszkiewicz <marcin@amiga.pl>

Index:rox-2002-05-26-selbutton/ROX-Filer/src/toolbar.c
*** rox-2002-05-26/ROX-Filer/src/toolbar.c	Mon May 27 09:46:34 2002
--- rox-2002-05-26-selbutton/ROX-Filer/src/toolbar.c	Mon May 27 10:00:31 2002
***************
*** 83,88 ****
--- 83,90 ----
  				   FilerWindow *filer_window);
  static void toolbar_sort_clicked(GtkWidget *widget,
  				   FilerWindow *filer_window);
+ static void toolbar_select_clicked(GtkWidget *widget,
+ 				   FilerWindow *filer_window);
  static GtkWidget *add_button(GtkWidget *bar, Tool *tool,
  				FilerWindow *filer_window);
  static GtkWidget *create_toolbar(FilerWindow *filer_window);
***************
*** 130,135 ****
--- 132,141 ----
  	 toolbar_sort_clicked, DROP_NONE, TRUE,
  	 NULL},
  	
+ 	{N_("Select"), GTK_STOCK_DND_MULTIPLE, N_("Select all/invert selection"),
+ 	 toolbar_select_clicked, DROP_NONE, TRUE,
+ 	 NULL},
+ 	
  	{N_("Details"), GTK_STOCK_JUSTIFY_LEFT, N_("Show extra details"),
  	 toolbar_details_clicked, DROP_NONE, TRUE,
  	 NULL},
***************
*** 375,380 ****
--- 381,404 ----
  	display_set_hidden(filer_window, !filer_window->show_hidden);
  }
  
+ static void toolbar_select_clicked(GtkWidget *widget, FilerWindow *filer_window)
+ {
+ 	GdkEvent	*event;
+ 
+ 	event = gtk_get_current_event();
+ 	if (event->type == GDK_BUTTON_RELEASE &&
+  	    ((GdkEventButton *) event)->button==2)
+ 	{
+ 		collection_invert_selection(filer_window->collection);
+ 	}
+ 	else if (event->type == GDK_BUTTON_RELEASE &&
+ 		 ((GdkEventButton *) event)->button==1)
+ 	{
+ 		collection_select_all(filer_window->collection);
+ 	}
+ 	filer_window->temp_item_selected = FALSE;
+ }
+ 
  static GtkWidget *create_toolbar(FilerWindow *filer_window)
  {
  	GtkWidget	*bar;
